### DISCUSSION TITLE
Is it safe to commit deployment.yaml into a public repository?

### DISCUSSION THREAD

**Discussion by mrzzy** [PrefectHQ/prefect#8249] [Q&A]
According to [Storage Blocks documentation](https://docs.prefect.io/tutorials/storage/#create-a-storage-block), secrets are obfuscated in the generated deployment.yaml.

Are any secrets embedded into deployment.yaml on `prefect deployment build -sb <SOME_BLOCK>`? If so, can be obfuscation be relied upon to protect them?

**Comment by ghost** [comment]
Seems safe to me if all values are referred to through secrets:
```
    access_token: '{{ prefect.blocks.secret.deployment-token }}'
```
